### PR TITLE
refactor: Convert NotificationChannelTest to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**
Difficult one. First time we needed `assertNotNull` (which was caught via a lint check).

There was difficulty in implementing `junit5` (known TODO on the Google Side), so I used the `kotlin-test` `assertNotNull`

### CODE REVIEW

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt [hunk 1]
```diff
@@ -13,88 +13,86 @@
...
-
+@KotlinCleanup("IDE Lint")
+@KotlinCleanup("JUnit 5")
+class NotificationChannelTest : InstrumentedTest() {
+    @get:Rule
+    var runtimePermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+    private var mCurrentAPI = -1
+    private var mTargetAPI = -1
+    private var mManager: NotificationManager? = null
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:49]: cleanup: lateinit

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt [hunk 2]
```diff
@@ -13,88 +13,86 @@
...
+            if (channel.name.toString().contains("LeakCanary")) {
+                expectedChannels += 1
             }
         }
-        assertEquals("Incorrect channel count", expectedChannels, channels.size());
-
-        for (NotificationChannels.Channel channel : NotificationChannels.Channel.values()) {
-            assertNotNull("There should be a reminder channel",
-                    mManager.getNotificationChannel(NotificationChannels.getId(channel)));
+        Assert.assertEquals(
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:86]: import `Assert.asserEquals` back

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt [hunk 3]
```diff
@@ -13,88 +13,86 @@
...
-    @Rule
-    public GrantPermissionRule mRuntimePermissionRule =
-            GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
-
-    private int mCurrentAPI = -1;
-    private int mTargetAPI = -1;
-    private NotificationManager mManager = null;
-
+@KotlinCleanup("IDE Lint")
+@KotlinCleanup("JUnit 5")
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:42]: I can't tell what this cleanup wants exactly. Could you explain it on the code?

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

**Review by BrayanDSO — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
LGTM - passes local re-integration test

**Pull request merged**